js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,69 @@
+// describing spec (what's being tested)
+describe("enhancedDate", function () {
+
+	// Test each method within object
+
+	// setDate
+	describe("setDate", function () {
+
+		it("Return same date object as date object given.", function () {
+			var newDate = new Date("Dec 12, 2015");
+			var newDateString = newDate.toString();
+			var setDateResult = enhancedDate.setDate(newDate);
+			var setDateString = setDateResult.toString();
+
+			expect(newDateString).to.equal(setDateString);
+		});
+
+		it("Give unix time then get date object back that represents that same time.", function () {
+			var dateNumber = 1432406605;
+			var setDateResult = enhancedDate.setDate(dateNumber);
+			var setDateNumber = setDateResult.valueOf();
+
+			expect(dateNumber).to.equal(setDateNumber);
```

Comment:
I’ve read the comments and I’m working my little brain as hard as I can. I used this example but I can’t quite connect the dots as to *how* it’s work. Can you tell me how the two objects(?) are being compared? What’s the difference between them?

To me it look like the first one is calling setDate method from the enhancedDate module and using the dateNumber of milliseconds supplied and comparing it to the plain old date object (not referencing the enhancedDate module) using the same number of milliseconds. If that’s right then were using a value we enter to compare it to a value returned by the enhancedDate module? Am I in the ballpark at all? Did that even make sense?

it(“Description here.", function () {
	var dateNumber = 1432406605;
	expect(enhancedDate.setDate(dateNumber)).to.deep.equal(new Date(dateNumber));
});




> On May 24, 2015, at 12:40 PM, jamdon <notifications@github.com> wrote:
> 
> In js/tests/enhancedDate.spec.js:
> 
> > + > +	it("Return same date object as date object given.", function () { > +	var newDate = new Date("Dec 12, 2015"); > +	var newDateString = newDate.toString(); > +	var setDateResult = enhancedDate.setDate(newDate); > +	var setDateString = setDateResult.toString(); > + > + expect(newDateString).to.equal(setDateString); > +	}); > + > +	it("Give unix time then get date object back that represents that same time.", function () { > +	var dateNumber = 1432406605; > +	var setDateResult = enhancedDate.setDate(dateNumber); > +	var setDateNumber = setDateResult.valueOf(); > + > +	expect(dateNumber).to.equal(setDateNumber); 
> This test works as well, but I would consider testing the method more directly. For example:
> expect(enhancedDate.setDate(dateNumber)).to.deep.equal(new Date(dateNumber);
> 
> —
> Reply to this email directly or view it on GitHub.
>